### PR TITLE
(PUP-5445) Add Arista support

### PR DESCRIPTION
**Pull request by LuvCurves**

### CODE REVIEW

**Comment by LuvCurves** [comment]
Requires this PR https://github.com/puppetlabs/puppet/pull/4372

File: acceptance/config/aio/options.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@
     'setup/common/pre-suite/000-delete-puppet-when-none.rb',
     'setup/aio/pre-suite/010_Install.rb',
     'setup/aio/pre-suite/015_PackageHostsPresets.rb',
+    'setup/aio/pre-suite/020_InstallCumulusModules.rb',
```
Comment (joshcooper) [inline acceptance/config/aio/options.rb]: looks like this file wasn't `git add`ed

File: acceptance/setup/aio/pre-suite/021_InstallAristaModule.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+platforms = hosts.map{|val| val[:platform]}
+skip_test "No arista hosts present" unless platforms.any? { |val| /eos/ =~ val }
```
Comment (joshcooper) [inline acceptance/setup/aio/pre-suite/021_InstallAristaModule.rb]: I think we prefer using `confine :to, :platform => /eos/`, though I defer to @johnduarte and @er0ck on that

**Comment by joshcooper** [comment]
@LuvCurves Could you updated the commit message to refer to the PUP ticket for this work? Usually we generate a set of tickets when adding a new platform. /cc @geoffnichols @melissa

**Comment by LuvCurves** [comment]
@joshcooper There is no Pup ticket for this work.  It is being tracked as a QA ticket.  This, like Cumulus will NOT be a platform that will be run in Puppet Agent CI, but a "NetDev" CI.

File: acceptance/setup/aio/pre-suite/021_InstallAristaModule.rb (hunk 2 above)
Comment (LuvCurves) [inline acceptance/setup/aio/pre-suite/021_InstallAristaModule.rb]: I need to do more than work with the master in this script, hence the lack of a confine.

File: acceptance/config/aio/options.rb (hunk 1 above)
Comment (LuvCurves) [inline acceptance/config/aio/options.rb]: Updated the options.rb

File: acceptance/setup/aio/pre-suite/021_InstallAristaModule.rb (hunk 2 above)
Comment (johnduarte) [inline acceptance/setup/aio/pre-suite/021_InstallAristaModule.rb]: @joshcooper the need is to execute work on masters and agents, but only if one or more of the agents is of type `eos`. This pattern is the best that @LuvCurves and I could come up with to identify those constraints.

**Comment by kylog** [comment]
@LuvCurves our release script (and TravisCI) will balk at the QA ticket reference (it looks for PUP tickets or one of "maint" or "docs" or "packaging"). To that end, can you change the commit summary to start with "(maint)"? Also, can you squash the commits?

Thanks!

**Pull request closed without merging**